scripts/portraits/batch_extract_portraits.py

```
@@ -0,0 +1,236 @@
+        output_path = output_dir / f"{filename}_{p['name']}_{offset_info}.png"
+        cv2.imwrite(str(output_path), portrait)
+
+    return True
+
+
+def _validate_path(path: Path) -> Path:
+    """验证并规范化路径，防止路径注入攻击"""
+    try:
+        resolved = path.resolve()
+    except RuntimeError as e:
+        raise ValueError(f"路径解析失败（可能存在循环符号链接）: {path}") from e
```

Ruff (0.16.2)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)